baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/StrictUnusedVariable.java

```diff
@@ -751,11 +748,12 @@ public Void visitClass(ClassTree tree, Void unused) {
             if (isSuppressed(tree)) {
                 return null;
             }
-            if (EXEMPTING_SUPER_TYPES.stream()
-                    .anyMatch(t ->
-                            isSubtype(getType(tree), Suppliers.typeFromString(t).get(state), state))) {
+
+            Pattern isRecord = Pattern.compile("record\\s+" + tree.getSimpleName() + "\\(");
+            if (isRecord.matcher(state.getSourceForNode(tree)).find()) {
```

Comment:
This could become expensive, I wonder if we can get away with `state.getSourceForNode(tree).startsWith("record ")`?

Reply:
Unfortunately records can have a visibility modifier so that check doesn't quite work